django/db/models/sql/query.py

```diff
@@ -855,8 +855,10 @@ def change_aliases(self, change_map):
                 if alias == old_alias:
                     table_aliases[pos] = new_alias
                     break
-        self.external_aliases = {change_map.get(alias, alias)
-                                 for alias in self.external_aliases}
+        self.external_aliases = {
+            change_map.get(alias, alias): (aliased or alias in change_map)
```

Comment:
If the table is already aliased (e.g. `JOIN table AS table_alias`) or it's being _changed_ and thus aliased.